R/read_csv.R

```diff
@@ -370,9 +388,124 @@ read_sample_csv <- function(files,
   read_cmdstan_csv(files, variables, sampler_diagnostics)
 }
 
+#' @rdname read_cmdstan_csv
+#' @export
+#' @param check_diagnostics Should diagnostic checks be performed? The default
+#'   is `TRUE` but set to `FALSE` to avoid checking diagnostics like
+#'   divergences.
+#'
+as_cmdstan_mcmc <- function(files, check_diagnostics = TRUE) {
+  csv_contents <- read_cmdstan_csv(files)
+  CmdStanMCMC2$new(csv_contents, files, check_diagnostics)
+}
+
+#' @rdname read_cmdstan_csv
+#' @export
+as_cmdstan_mle <- function(files) {
+  csv_contents <- read_cmdstan_csv(files)
+  CmdStanMLE2$new(csv_contents, files)
+}
+
+#' @rdname read_cmdstan_csv
+#' @export
+as_cmdstan_vb <- function(files) {
+  csv_contents <- read_cmdstan_csv(files)
+  CmdStanVB2$new(csv_contents, files)
+}
+
+
 
 # internal ----------------------------------------------------------------
 
+# CmdStanFit2 -------------------------------------------------------------
+#' Create a CmdStanMCMC/MLE/VB-ish objects from `read_cmdstan_csv()` output
+#' instead of from a CmdStanRun object
+#'
+#' The resulting object has fewer methods than a CmdStanMCMC/MLE/VB object
+#' because it doesn't have access to the CSV files directly.
+#'
+#' @noRd
+#'
+CmdStanMCMC2 <- R6::R6Class(
```

Comment:
I put these in the read_csv.R file because of the use case but we could also but these in the fit.R file with the definition of CmdStanMCMC, CmdStanMLE, etc. Preference?

Reply:
I think having them here is fine.